frame/support/src/storage/bounded_vec.rs

```diff
@@ -159,6 +159,21 @@ impl<T, S> BoundedVec<T, S> {
 	pub fn pop(&mut self) -> Option<T> {
 		self.0.pop()
 	}
+
+	/// Exactly the same semantics as [`Vec::iter`].
+	pub fn iter(&mut self) -> core::slice::Iter<'_, T> {
```

Comment:
```suggestion
	pub fn iter(&self) -> core::slice::Iter<'_, T> {
```
Sorry for the DQ but it is already possible to call `iter()` and `into_iter()` on a bounded vec.  
Maybe from the `IntoIterator` impl?  
The `iter_mut` is the only one missing, but I think implementing `DerefMut` is enough to make it work:  
```rust
impl<T, S> core::ops::DerefMut for BoundedVec<T, S> {
    fn deref_mut(&mut self) -> &mut Self::Target {
		&mut self.0
	}
}
```

Reply:
That's no good since there's nothing stopping you from using it to push more items than the bound would allow.

Reply:
`DerefMut` is not implemented on purpose, because then you can just modify the underlying vector without checking for bounds etc.

Reply:
Oh yes...  

Most changes here probably carry over fo `WeakBoundedVec`, maybe worth to add it there as well before we touch it twice.